src/broker/cas_cgw.c

```diff
@@ -1025,14 +1019,31 @@ cgw_error_msg (SQLHANDLE hHandle, SQLSMALLINT hType, RETCODE retcode)
 void
 cgw_database_disconnect ()
 {
-  cgw_cleanup_handle (local_odbc_handle);
+  if (local_odbc_handle == NULL)
+    {
+      return;
+    }
+
+  if (local_odbc_handle->hstmt)
+    {
+      SQLFreeHandle (SQL_HANDLE_STMT, local_odbc_handle->hstmt);
+      local_odbc_handle->hstmt = NULL;
+    }
+
+  if (local_odbc_handle->hdbc)
+    {
+      SQLDisconnect (local_odbc_handle->hdbc);
+      SQLFreeHandle (SQL_HANDLE_DBC, local_odbc_handle->hdbc);
+      local_odbc_handle->hdbc = NULL;
+    }
 
-  is_database_connected = -1;
+  connected_db_name[0] = '\0';
+  connected_db_user[0] = '\0';
+  connected_db_passwd[0] = '\0';
```

Comment:
```suggestion
  connected_db_name[0] = '\0';
  connected_db_user[0] = '\0';
  connected_db_passwd[0] = '\0';
```
how about memset( local_odbc_handle, 0, sizeof(local_odbc_handle) );